src/utilities/api-requests.ts

```diff
@@ -10,13 +15,23 @@ import axios from "axios";
  */
 export async function request<T>(uri: string, defaultValue: T): Promise<T> {
     let data = defaultValue;
-    await axios
-        .get(uri)
-        .then((res) => {
-            data = res.data as T;
-        })
-        .catch((error) => {
-            console.error(`Request to ${uri} failed with\n${error}`);
-        });
+    let attempt = 0;
```

Comment:
Given that we are using Axios, why not use [axios-retry](https://github.com/softonic/axios-retry) to handle this kind of logic instead?

Reply:
If we do not use axios-retry, how about storing the error-ed results in an array and doing a .length on it? And possibly logging the error results to console at the end?

We might even use this information later for a toast explaining to the user that there is some connection issue/rate limiting issue.

Reply:
I think it is nice to be able to handle the different codes manually. It turns out the 429 error returns the time the lock is lifted in the header. So we can parse the header and wait the exact time for the API to unlock. We can also handle other error codes so they do not wait as long before retrying.

Reply:
Neat!